app/Repositories/PostRepository.php

```diff
@@ -77,6 +79,21 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
 
         $signup = Signup::find($signupId);
 
+        // Get the action_id either from the payload or the DB.
+        if (isset($data['action_id'])) {
+            $actionId = $data['action_id'];
+        } else {
+            $type = isset($data['type']) ? $data['type'] : 'photo';
+            $action = Action::where('campaign_id', $signup->campaign_id)->where('post_type', $type)->where('name', $data['action'])->first();
+
+            if (! $action) {
+                // Throw an exception that no action exists... should this be caught at an earlier point though?
+                throw new ModelNotFoundException('Action not found.');
```

Comment:
Is there a better exception that we should throw here? This might be too rigid - if an action doesn't exist in the database and anything gets out of sync (an action isn't created first for a post type and isn't added in Contentful for example) creating a post will throw an error and might cause some issues.

Reply:
@chloealee can you explain the syncing issues again? I'm not sure I follow

Reply:
@katiecrane of course! For now, we won't require that `action_id` is sent along in the payload when a post is being created via the `POST v3/api/posts` endpoint (we will do this later but not immediately to allow other systems to adjust). At the same time, we don't want anything to break and have all work exactly the same so no systems should really see a difference with this PR. 

To account for this, we first will try to get the `action_id` from the payload if it is sent. If it is not, we'll try to get the `action` from the `actions` table in Rogue with this query, which we should be able to do with the `name` of the action, the `campaign_id` and `post_type`. This is mostly for existing campaigns so people can still reportback! 

However, if a campaign lead forgets to create an `action` on a new campaign for a new type of post, this will break. It should work with all other existing campaigns since we will run a backfill to migrate all the existing actions to the `actions` table and update all existing posts to have the correct `action_id` (scripts in [this PR](https://github.com/DoSomething/rogue/pull/825). 

Not sure if this is the best way to handle this error though so trying to think through if there is a better way and will make it easier to debug and not lose any reportbacks...

Reply:
@chloealee Thank you for the easy to understand description! I think it makes sense to keep it this way, it sounds like we'd only ever be missing an action in the case where it is a new action that wasn't properly created, and in that case we should be throwing an error!

Reply:
Sounds good, thanks @katiecrane ! I'll remove that comment then now :)